packages/openapi3/src/schema-emitter.ts

```diff
@@ -936,10 +936,14 @@ export function getDefaultValue(program: Program, defaultType: Value): any {
       return null;
     case "EnumValue":
       return defaultType.value.value ?? defaultType.value.name;
+    case "ScalarValue":
+      return serializeValueAsJson(program, defaultType, defaultType.type);
```

Comment:
I think you should just be able to replace the whole function content with just calling that

Reply:
and remove the `invalid-default` diagnostic in lib.ts